spec/support/tempfile.cr

```diff
@@ -0,0 +1,43 @@
+require "tempfile"
+require "file_utils"
+
+SPEC_TEMPFILE_PATH = File.join(Tempfile.dirname, "crystal-spec-#{Random.new.hex(8)}")
```

Comment:
If you change this to `hex(4)` does it pass osx without any changes to `bin/ci`? What about if you change this to `s/crystal-spec/cr-spec`? If those two are enough then we can sidestep this whole stalemate.